src/tools/genpolicy/src/policy.rs

```diff
@@ -737,31 +771,103 @@ impl AgentPolicy {
             resource.get_annotations(),
             service_account_name,
         );
+        debug!(
+            "get_container_process: after get_env_variables: User = {:?}",
+            &process.User
+        );
 
         substitute_env_variables(&mut process.Env);
+        debug!(
+            "get_container_process: after substitute_env_variables: User = {:?}",
+            &process.User
+        );
+
         substitute_args_env_variables(&mut process.Args, &process.Env);
+        debug!(
+            "get_container_process: after substitute_args_env_variables: User = {:?}",
+            &process.User
+        );
 
-        c_settings.get_process_fields(&mut process);
+        ///////////////////////////////////////////////////////////////////////////////////////
+        // genpolicy-settings.json information.
+        let v1_policy = self
+            .config
+            .settings
+            .cluster_config
+            .pause_container_id_policy
+            == "v1";
+        if !v1_policy {
+            let v2_policy = self
+                .config
+                .settings
+                .cluster_config
+                .pause_container_id_policy
+                == "v2";
+            if !v2_policy {
+                panic!(
+                    "Unsupported pause_container_id_policy = {} - must be v1 or v2 in the settings file",
+                    self.config.settings.cluster_config.pause_container_id_policy
+                );
+            }
+        }
+        let update_additional_gids = !is_pause_container || v1_policy;
```

Comment:
I wonder why this update_additional_gids logic only applies to the old pause container image? Shouldn't the outcome of additionalGids be handled by the contents of the container image layers (/etc/passwd, /etc/groups)?